### ISSUE TITLE
Error launching source instance: timeout while waiting for state to become 'success'. last error: %!s(<nil>)

### ISSUE THREAD

**Issue by kieferrj** [hashicorp/terraform#8857]
Hey guys - this is an easy one. It looks like when theres no more capacity for an instance type in an AZ, and TF tries to make an instance and fails because of it, it fails to pass that message down through TF, and all you get is the subject line instead.

I was able to discover the nature of the error by attempting to replicate what TF was doing manually in the WebUI. The WebUI returned the proper error message: "We currently do not have sufficient (instance_type (m4.large, in my scenario)) capacity in the Availability Zone you requested (AZ). Our system will be working on provisioning additional capacity. You can currently get a (instance_type) by not specifying an Area Zone in your request or choosing (accounts_other_AZs)."

This is in version 0.7.0

**Comment by jen20** [comment]
Hi @kieferrj! This definitely looks like a failure to handle the specific error type. Unfortunately it's also very hard to test for since AWS don't run out of capacity that often or in a deterministic fashion! We'll investigate this since there is clearly a bug given the error message construction. Thanks for opening an issue!

**Comment by akoeb-zalando** [comment]
requesting m3.large in eu-central-1a currently produces this error (and did so in the past).

running with TF_LOG=debug, I can see following response from AWS:

```
2016/09/20 11:54:43 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/09/20 11:54:43 [DEBUG] plugin: terraform: aws-provider (internal) 2016/09/20 11:54:43 [DEBUG] [aws-sdk-go] DEBUG: Response ec2/RunInstances Details:
2016/09/20 11:54:43 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/09/20 11:54:43 [DEBUG] plugin: terraform: HTTP/1.1 500 Internal Server Error
2016/09/20 11:54:43 [DEBUG] plugin: terraform: Connection: close
2016/09/20 11:54:43 [DEBUG] plugin: terraform: Transfer-Encoding: chunked
2016/09/20 11:54:43 [DEBUG] plugin: terraform: Date: Tue, 20 Sep 2016 09:54:43 GMT
2016/09/20 11:54:43 [DEBUG] plugin: terraform: Server: AmazonEC2
2016/09/20 11:54:43 [DEBUG] plugin: terraform: 
2016/09/20 11:54:43 [DEBUG] plugin: terraform: 1fa
2016/09/20 11:54:43 [DEBUG] plugin: terraform: <?xml version="1.0" encoding="UTF-8"?>
2016/09/20 11:54:43 [DEBUG] plugin: terraform: <Response><Errors><Error><Code>InsufficientInstanceCapacity</Code><Message>We currently do not have sufficient m3.large capacity in the Availability Zone you requested (eu-central-1a). Our system will be working on provisioning additional capacity. You can currently get m3.large capacity by not specifying an Availability Zone in your request or choosing eu-central-1b.</Message></Error></Errors><RequestID>5cbe8339-9fed-49d7-bde0-d4b1c15eb1e3</RequestID></Response>
2016/09/20 11:54:43 [DEBUG] plugin: terraform: 0
2016/09/20 11:54:43 [DEBUG] plugin: terraform: 
2016/09/20 11:54:43 [DEBUG] plugin: terraform: 
2016/09/20 11:54:43 [DEBUG] plugin: terraform: -----------------------------------------------------

```

**Comment by Bowbaq** [comment]
Ran into this today. It seems that in addition to failing the terraform run, in some cases AWS still provisions the instance. This causes a subsequent call to `terraform destroy` to fail since security groups have "dependent objects" that aren't tracked in the state file.

**Comment by archbungle** [comment]
I'm getting this with TF 0.8.8 and 0.9.1. With r4.16xlarge instances, regardless of whether I try launch 1 or 10 the deploy fails with timeout waiting for instance. Is there a workaround for this ?

**Comment by twelcome** [comment]
Some additional notes:

- I can consistently launch 10 r4.8xlarge instances ok
- I get consistent timeouts with launching any number of r4.16xlarge instances.

Definitely seems related to instance deploy time - if there was a relieable tunable in TF for this I'm sure the problem could be resolved.

**Comment by radeksimko** [comment]
Hi folks,
we have improved the error message by adding the state reason, https://github.com/hashicorp/terraform/pull/14479 was just merged and will be part of the next release (0.9.6), therefore I'm closing this issue.

Let me know if you have any suggestions on further improvement.
